src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -0,0 +1,3926 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    public class TargetTypedObjectCreationTests : CSharpTestBase
+    {
+        private static readonly CSharpParseOptions TargetTypedObjectCreationTestOptions = TestOptions.RegularPreview;
+
+        private static CSharpCompilation CreateCompilation(string source, CSharpCompilationOptions options = null, IEnumerable<MetadataReference> references = null)
+        {
+            return CSharpTestBase.CreateCompilation(source, options: options, parseOptions: TargetTypedObjectCreationTestOptions, references: references);
+        }
+
+        [Fact]
+        public void TestInLocal()
+        {
+            var source = @"
+using System;
+
+struct S
+{
+}
+
+class C
+{
+    public static void Main()
+    {
+        C v1 = new();
+        S v2 = new();
+        S? v3 = new();
+        Console.Write(v1);
+        Console.Write(v2);
+        Console.Write(v3);
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+
+            CompileAndVerify(comp, expectedOutput: "CSS");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes().OfType<ImplicitObjectCreationExpressionSyntax>().ToArray();
+
+            assert(0, type: "C", convertedType: "C", symbol: "C..ctor()", ConversionKind.Identity);
+            assert(1, type: "S", convertedType: "S", symbol: "S..ctor()", ConversionKind.Identity);
+            assert(2, type: "S", convertedType: "S?", symbol: "S..ctor()", ConversionKind.ImplicitNullable);
+
+            void assert(int index, string type, string convertedType, string symbol, ConversionKind conversionKind)
+            {
+                var @new = nodes[index];
+                Assert.Equal(type, model.GetTypeInfo(@new).Type.ToTestDisplayString());
+                Assert.Equal(convertedType, model.GetTypeInfo(@new).ConvertedType.ToTestDisplayString());
+                Assert.Equal(symbol, model.GetSymbolInfo(@new).Symbol.ToTestDisplayString());
+                Assert.Equal(conversionKind, model.GetConversion(@new).Kind);
+            }
+        }
+
+        [Fact]
+        public void TestInExpressionTree()
```

Comment:
@jcouv what are your thoughts on this? On the one hand, this feature needs nothing new to work in expression trees. On the other hand, we haven't really added anything new to expression trees in a while, and it'd be kind of weird that you can use target-typed new but not `??` in them.

Reply:
We have precedent with the target-typed `default` feature. It too works in expression trees, because it's just a `default(...)` expression and we didn't distinguish the two.

As long as expression trees don't need anything new to represent what we need, then it's fine to represent new features (we don't need to block for the purpose of blocking).

Reply:
> As long as expression trees don't need anything new to represent what we need, then it's fine to represent new features (we don't need to block for the purpose of blocking).

Completely agree. We generally do not base our decision about supporting things in expression trees on syntactic forms or on whether other (unrelated) feature are supported.